workflow/cron/operator.go

```diff
@@ -115,8 +119,54 @@ func getWorkflowObjectReference(wf *v1alpha1.Workflow, runWf *v1alpha1.Workflow)
 func (woc *cronWfOperationCtx) persistUpdate() {
 	_, err := woc.cronWfIf.Update(woc.cronWf)
 	if err != nil {
-		woc.log.WithError(err).Error("failed to update CronWorkflow")
+		if errors.IsConflict(err) {
+			reapplyErr := woc.reapplyUpdate()
+			if reapplyErr != nil {
+				woc.log.WithError(reapplyErr).WithField("original error", err).Error("failed to update CronWorkflow after reapply attempt")
+			}
+		} else {
+			woc.log.WithError(err).Error("failed to update CronWorkflow")
+		}
+	}
+}
+
+func (woc *cronWfOperationCtx) reapplyUpdate() error {
+	orig, err := json.Marshal(woc.origCronWf)
+	if err != nil {
+		return err
+	}
+	curr, err := json.Marshal(woc.cronWf)
+	if err != nil {
+		return err
+	}
+	patch, err := jsonpatch.CreateMergePatch(orig, curr)
+	if err != nil {
+		return err
+	}
+	for i := 0; i < 5; i++ {
+		currCronWf, err := woc.cronWfIf.Get(woc.name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		currCronWfBytes, err := json.Marshal(currCronWf)
+		if err != nil {
+			return err
+		}
+		newCronWfBytes, err := jsonpatch.MergePatch(currCronWfBytes, patch)
+		if err != nil {
+			return err
+		}
+		var newCronWf v1alpha1.CronWorkflow
+		err = json.Unmarshal(newCronWfBytes, &newCronWf)
+		if err != nil {
+			return err
+		}
+		_, err = woc.cronWfIf.Update(&newCronWf)
+		if err == nil {
+			return nil
```

Comment:
do we want to return the error on attempt 5?

Reply:
We do. After attempt 5 the loop breaks and returns an error on line 169

Reply:
sorry - not "an error" - "the error" i.e. the error returned by `Update`

Reply:
Sure, I see what you mean. Will refactor